app/controllers/rooms_controller.rb

```diff
@@ -1,115 +1,167 @@
-    @room_id = room
-    @issuesForRoom = Issue.where("room_id = ?", room)
-
-    @issues = Array.new
-
-    @issuesForRoom.each do |issue| 
-      issueEntity = IssueEntity.new
-      type = IssueType.where("id = ?", issue.issue_type_id).first
-      issueEntity.issue_type = issue.issue_type_id
-      issueEntity.room_id =@room
-      issueEntity.icon_path = type.icon_path
-      issueEntity.is_done = issue.is_done
-      issueEntity.issue_description = type.issue_description
-
-      @issues.push(issueEntity)
+    @room_id = params[:room_id]
+    @issues = create_issue_entities(@issues_for_room)
+  end
+
+  def create_issue_entities(issues_for_room)
+    issues = []
+    issues_for_room.each do |issue|
+      type = IssueType.where('id = ?', issue.issue_type_id).first
+      issue_entity = IssueEntity.new(issue.issue_type_id, params[:room_id],
+                                     issue.is_done, type.issue_description,
+                                     type.icon_path)
+      issues.push(issue_entity)
     end
+    return issues
   end
 
   def report_issue
     room_id = params[:room][:room_id]
-    issueType = params[:room][:issue_type]
-    @issue = Issue.where("room_id = ? AND issue_type_id = ?", room_id, issueType).first
-    @room = Room.where("room_id = ?", room_id).first
-    currentValue = @issue.is_done
-    @issue.update_attribute(:is_done, !currentValue)
-    @issue.update_attribute(:fix_comment, params[:room][:comment])
-    @issue.update_attribute(:completion_date, Date.today)
-    @room.update_attribute(:is_clean, false)
-
+    issue_type = params[:room][:issue_type]
+    issue = Issue.where('room_id = ? AND issue_type_id = ?',
+                        room_id, issue_type).first
+    room = Room.where('room_id = ?', room_id).first
+    current_value = issue.is_done
+    issue.update_attributes(is_done: !current_value,
+                            fix_comment: params[:room][:comment],
+                            completion_date: Date.today)
+    room.update_attribute(:is_clean, false)
     redirect_back(fallback_location: root_path)
   end
+
+  # Helper methods
+
+  def fetch_issues_by_room_id
+    @issues_for_room = Issue.where('room_id = ?', params[:room_id])
+  end
+
+  def fetch_all_rooms
+    @rooms = Room.all
+  end
+
+  def fetch_selected_issues
+    @selected_issues = Issue.where(room_id: params[:id])
+  end
+
+  def fetch_all_issue_types
+    @issues = IssueType.all
+  end
+
+  def fetch_room_by_id
+    @room = Room.find(params[:id])
+  end
+
+  def add_new_issues_to_room(selected_issues, id)
+    selected_issues.each do |issue|
+      unless Issue.exists?(room_id: id, issue_type_id: issue.to_i)
+        Issue.create(room_id: id.to_i, issue_type_id: issue.to_i,
+                     priority: 'Medium', is_done: true,
+                     is_recurring: false)
+      end
+    end
+  end
+
+  def remove_old_issues_from_room(selected_issues, id)
+    @issues_for_room = Issue.where('room_id = ?', id)
```

Comment:
The fetch_issues_by_room_id is a before_action fetch - it's OK to use it in a view where I need a pre-loaded array/value etc. but I don't see the point in calling it in a re-usable method. We have the ID passed by argument as there is no guarantee that in every place I call it - the room ID will be conveniently stored in the params[] table and will always by called :room_id.